### ISSUE TITLE
run benchmark tests

### ISSUE THREAD

**Issue by rafapereirabr** [ipeaGIT/r5r#106]
It would be important to run some benchmark tests and compare the travel time estimates from `r5r` against the results from some other routing packages such as:
- opentripplanner
- dodgr
- stplanr
- ArcGis
- OTP via python scripts

It would also be nice to have a look at how the computation time varies across these packages and perhaps mention this comparison on the `readme.md` file

**Comment by dhersz** [comment]
It might be interesting adding Google Maps API to this list, since it offers both [Directions and Distance Matrix](https://cloud.google.com/maps-platform/routes), their version of detailed_itineraries and travel_time_matrix.

**Comment by kauebraga** [comment]
- Use São Paulo's system for tests

**Comment by kauebraga** [comment]
## Tests results for São Paulo's GTFS, comparing OpenTripPlannerPy and R5

### Resources
- CPU: Intel Xeon 2.30 GHz
- RAM: 161 GB
- Number of cores used in parallel processing: 12

### System information
- Windows Server 2012 R2 x64
- R version 4.0.2 (2020-06-22)

### Parameters
- OD points: 10124 x 10124 = 102,495,376 pairs
- Departure time: 2020-01-29 06:00:00
- Max trip duration: 120 minutes

### Results (OTPPy x R5, time in minutes)

| step         | otp_time | r5_time |
|--------------|----------|---------|
| load graph   | 2        | 0.25    |
| ttime matrix | 77       | 3.5     |
|              |          |         |

**Comment by kauebraga** [comment]
In a general way, here is the comparison of the overall elapsed time (graph loading and travel time matrix estimation) with the same resources/system/parameters (except for the OD points) for every city included in the [Access to Opportunities project](https://www.ipea.gov.br/acessooportunidades/en/). Time in minutes:

| muni           | od_points | time_elapsed_otp | time_elapsed_r5 |
|----------------|-----------|------------------|-----------------|
| Fortaleza      |      2269 |              5.9 |             0.6 |
| São Paulo      |     10124 |             77.4 |             3.5 |
| Rio de Janeiro |      6840 |             18.4 |             1.6 |
| Curitiba       |      3939 |              6.7 |             1.0 |
| Porto Alegre   |      3340 |              7.9 |             0.8 |
| Belo Horizonte |      2667 |              9.3 |             0.8 |
| Recife         |      1448 |              2.8 |             0.3 |

**Comment by rafapereirabr** [comment]
This is a huge improvment in computation speed. @kauebraga , have you had a chance of looking into differences in travel-time estimates? Are these differences larger for larger trips in particular?

**Comment by kauebraga** [comment]
## Comparison between travel-time estimates for each OD pair

The graph below compares the distribution of the log difference between R5 and OTP travel-time estimates (each point is an OD pair). Values were truncated to the [-0.5, 0.5] for readability. Points above y=0 indicates where R5 overestimates travel times in relation to OTP:
![comp_boxplot](https://user-images.githubusercontent.com/22420248/92634207-50ab7f00-f2aa-11ea-9631-7a22e21e397f.png)
.  
Summary table:

| sigla_muni | dif_mean | dif_median | dif_max | dif_min | nrow_r5    | nrow_otp   |
|------------|----------|------------|---------|---------|------------|------------|
| for        |     1.2% |      -0.6% |  216.8% | -213.3% |  4977562.0 |  5012689.0 |
| spo        |   -12.2% |     -11.4% |  280.6% | -268.7% | 28290470.0 | 46143997.0 |
| rio        |   -15.7% |     -13.0% |  297.1% | -299.0% | 19341294.0 | 15646364.0 |
| cur        |    -0.5% |      -0.8% |  292.5% | -262.4% | 12779630.0 | 12866935.0 |
| poa        |     1.6% |       0.8% |  286.7% | -282.8% |  9194755.0 |  9455256.0 |
| bho        |     9.3% |       8.0% |  292.2% | -260.6% |  5752772.0 |  6424157.0 |
| rec        |    -5.6% |      -2.6% |  282.0% | -282.2% |  1347519.0 |  1396739.0 |

### How does this difference relate to travel distance?
Using distances between OD pairs calculated with the `dodgr` package we can see that higher values of relative difference are concentrated around short-distanced OD pairs. 

![comp_dist](https://user-images.githubusercontent.com/22420248/92637916-e39ae800-f2af-11ea-917d-c0f293005c31.png)

**Comment by rafapereirabr** [comment]
For the record, here is a [gist with a reproducible code to compare the performance of `r5r` and `opentripplanne` packages](https://gist.github.com/dhersz/14c4b7927a9ef91aeddcc3bf8bd0712a). The code uses the sample data stored in the r5r package.

**Comment by rafapereirabr** [comment]
Christopher D. Higgins et al have [a new paper running a benchmark comparing many routing engines](https://www.researchgate.net/profile/Christopher-Higgins-5/publication/355721229_Calculating_Place-Based_Transit_Accessibility_Methods_Tools_and_Algorithmic_Dependence/links/617b5a0aa767a03c14c6d563/Calculating-Place-Based-Transit-Accessibility-Methods-Tools-and-Algorithmic-Dependence.pdf).
